test/cqlpy/test_tablets.py

```diff
@@ -181,7 +181,38 @@ def test_describe_initial_tablets(cql, skip_without_tablets):
             "AND TABLETS = { 'initial' : 1 }"
     with new_test_keyspace(cql, ksdef) as keyspace:
         desc = cql.execute(f"DESCRIBE KEYSPACE {keyspace}")
-        assert "and tablets = {'initial': 1}" in desc.one().create_statement.lower()
+        assert "and tablets = {'enabled': true, 'initial': 1}" in desc.one().create_statement.lower()
+
+
+# Test to ensure metadata is printed when tablets are enabled during schema description, regardless of the use of WITH INTERNALS.
+#
+# Reproduces https://github.com/scylladb/scylladb/issues/22866
+@pytest.mark.parametrize("with_internals", [True, False])
+@pytest.mark.parametrize("tablets", [None, True, False])
+@pytest.mark.parametrize("initial", [None, 0, 7])
+def test_describe_tablets_internals(cql, with_internals, tablets, initial):
+    tablets_str = (f"'enabled': {tablets}" if tablets is not None else "").lower()
+    initial_str = f"'initial': {initial}" if initial is not None else ""
+
+    tablets_def = ""
+    if tablets_str or initial_str:
+        tablets_def = " AND TABLETS = {"
+        tablets_def += ", ".join(filter(None, [tablets_str, initial_str]))
+        tablets_def += "}"
+    ksdef = f"WITH REPLICATION = {{ 'class' : 'NetworkTopologyStrategy', 'replication_factor' : '1' }}{tablets_def}"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        internals = " WITH INTERNALS" if with_internals else ""
+        desc = cql.execute(f"DESCRIBE SCHEMA{internals};")
+        results = desc.all()
```

Comment:
I don't think this line is needed at all - I think you can do `for res in desc` directly.
Or am I missing something?